problem building `ponyc` if an older version of `ponyc` is installed

## System Information

- Mac OS X 10.11.6
- LLVM 3.9

## Description

If you have an older version of `ponyc` installed it can cause problems building a new version. In my case I was getting errors like this:

```
Linking ponyc
undef: __ZN4llvm19SmallPtrSetImplBaseD2Ev
undef: __ZN4llvm18PassManagerBuilder12addExtensionENS0_16ExtensionPointTyEPFvRKS0_RNS_6legacy15PassManagerBaseEE
undef: __ZN4llvm9DIBuilder20createSubroutineTypeENS_14DITypeRefArrayEj
undef: __ZN4llvm9DIBuilder12createMethodEPNS_7DIScopeENS_9StringRefES3_PNS_6DIFileEjPNS_16DISubroutineTypeEbbjjPNS_6DITypeEjbNS_24MDTupleTypedArrayWrapperINS_19DITemplateParameterEEE
undef: _LLVMGetTargetMachineData
undef: __ZN4llvm9DIBuilder17createCompileUnitEjNS_9StringRefES1_S1_bS1_jS1_NS0_17DebugEmissionKindEyb
Undefined symbols for architecture x86_64:
  "llvm::SmallPtrSetImplBase::~SmallPtrSetImplBase()", referenced from:
      HeapToStack::canStackAlloc(llvm::Instruction*, llvm::DominatorTree&, llvm::SmallVector<llvm::CallInst*, 4u>&, llvm::SmallVector<llvm::Instruction*, 4u>&) in libponyc.a(genopt.o)
      HeapToStack::canBeReused(llvm::Instruction*, llvm::Instruction*, llvm::DominatorTree&) in libponyc.a(genopt.o)
  "llvm::PassManagerBuilder::addExtension(llvm::PassManagerBuilder::ExtensionPointTy, void (*)(llvm::PassManagerBuilder const&, llvm::legacy::PassManagerBase&))", referenced from:
      _genopt in libponyc.a(genopt.o)
  "llvm::DIBuilder::createSubroutineType(llvm::DITypeRefArray, unsigned int)", referenced from:
      _LLVMDIBuilderCreateSubroutineType in libponyc.a(gendebug.o)
      _make_function_debug in libponyc.a(genfun.o)
     (maybe you meant: __ZN4llvm9DIBuilder20createSubroutineTypeENS_14DITypeRefArrayEjj)
  "llvm::DIBuilder::createMethod(llvm::DIScope*, llvm::StringRef, llvm::StringRef, llvm::DIFile*, unsigned int, llvm::DISubroutineType*, bool, bool, unsigned int, unsigned int, llvm::DIType*, unsigned int, bool, llvm::MDTupleTypedArrayWrapper<llvm::DITemplateParameter>)", referenced from:
      _LLVMDIBuilderCreateMethod in libponyc.a(gendebug.o)
      _make_function_debug in libponyc.a(genfun.o)
  "_LLVMGetTargetMachineData", referenced from:
      _codegen in libponyc.a(codegen.o)
  "llvm::DIBuilder::createCompileUnit(unsigned int, llvm::StringRef, llvm::StringRef, llvm::StringRef, bool, llvm::StringRef, unsigned int, llvm::StringRef, llvm::DIBuilder::DebugEmissionKind, unsigned long long, bool)", referenced from:
      _codegen in libponyc.a(codegen.o)
      _LLVMDIBuilderCreateCompileUnit in libponyc.a(gendebug.o)
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [build/release/ponyc] Error 1
```

This seems to happen because `-L /usr/local/lib` (added in d59c185efd7eb38298147943d337ecbc17d889e3) comes before the local build directory, so the installed versions of `libponyc.a` and `libponyrt.a` are used rather than the ones that are created as part of the current build.

## Steps To Reproduce

1. Checkout and install an older version of `ponyc` using `sudo make install`.
2. Checkout the latest version of `ponyc` and build it using `make`

## Expected Result

You will get a build of `ponyc`.

## Actual Result

You get errors.

## Notes

@SeanTAllen believes he has a fix for this that involves moving the `-L /usr/local/lib` argument to a position after the local build directory in the arguments that are passed to the compiler.

--- a/Makefile
+++ b/Makefile
@@ -90,7 +90,7 @@ destdir ?= $(prefix)/lib/pony/$(tag)
 LIB_EXT ?= a
 BUILD_FLAGS = -march=$(arch) -mtune=$(tune) -Werror -Wconversion \
   -Wno-sign-conversion -Wextra -Wall
-LINKER_FLAGS = -march=$(arch) -mtune=$(tune) -L /usr/local/lib
+LINKER_FLAGS = -march=$(arch) -mtune=$(tune)
 AR_FLAGS ?= rcs
 ALL_CFLAGS = -std=gnu11 -fexceptions \
   -DPONY_VERSION=\"$(tag)\" -DLLVM_VERSION=\"$(llvm_version)\" \
@@ -610,7 +610,7 @@ define CONFIGURE_LINKER_WHOLE
 endef
 
 define CONFIGURE_LINKER
-  $(eval linkcmd := $(LINKER_FLAGS) -L $(lib))
+  $(eval linkcmd := $(LINKER_FLAGS) -L $(lib) -L /usr/local/lib )
   $(eval linker := $(CC))
   $(eval libs :=)
 

